### ISSUE TITLE
Style/StringLiterals doesn't correctly handle strings with non-ASCII characters in them

### ISSUE THREAD

**Issue by eyusupov** [bbatsov/rubocop#3024]
When the string contains a non-ASCII character (for example, "Curaçao"), the cop treats it as a case where double quotes are necessary.  

## Expected behavior
The cop should detect unnecessary double quotes. 

## Actual behavior
The cop does not detect them.

## Steps to reproduce the problem
Run RuboCop on a file containing a line like this:
```ruby
puts "Curacao"
puts "Curaçao"
```
Notice that only first line results in a style violation.

## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.16.0 (using Parser 2.1.2, running on ruby 2.0.0 x86_64-darwin12.4.0)
```

**Comment by mikegee** [comment]
Dupe of https://github.com/bbatsov/rubocop/issues/3017

Also, your version of rubocop is very old. Please verify the problem with the latest version before opening an issue.

**Comment by eyusupov** [comment]
Oops, looks like I've pasted the wrong version. Thanks for noticing that it's dupe @mikegee
